lib/python/vdm/server/HTTPListener.py

```diff
@@ -480,6 +483,41 @@ def map_deployment_users(request, user):
     return deployment_user[0]
 
 
+def ignore_sighup():
+    signal.signal(signal.SIGHUP, signal.SIG_IGN)
+    signal.signal(signal.SIGINT, signal.SIG_IGN)
+
+def start_local_server(deploymentcontents):
+    filename = os.path.join(PATH, 'deployment.xml')
+    deploymentfile = open(filename, 'w')
+    deploymentfile.write(deploymentcontents)
+    deploymentfile.close()
+    voltdb_dir = os.path.realpath(os.path.join(MODULE_PATH, '../../../..', 'bin'))
+    voltdb_cmd = [ 'nohup', os.path.join(voltdb_dir, 'voltdb'), 'create', '-d', filename ]
+    outfilename = os.path.join(PATH, 'voltserver.output')
+    outfile = open(outfilename, 'w')
+
+    # Start server in a separate process
+    voltserver = subprocess.Popen(voltdb_cmd, stdout=outfile, stderr=subprocess.STDOUT,
+                                  preexec_fn=ignore_sighup)
+
+    initialized = False
+    rfile = open(outfilename, 'r')
+
+    # Wait till server is ready or process exited due to error.
+    # Wait for a couple of seconds to see if the server errors out
+    endtime = time.time() + 2
+    while ((endtime-time.time()>0) and
+           (voltserver.returncode is None) and (not initialized)):
+        time.sleep(0.5)
+        voltserver.poll()
+	initialized = 'Server completed initialization' in rfile.readline()
```

Comment:
nice to know. thanks for that bit of info. are you closing the file when you hit it?